gate-web/src/main/groovy/com/netflix/spinnaker/gate/config/PostConnectionConfiguringJedisConnectionFactory.java

```diff
@@ -0,0 +1,73 @@
+package com.netflix.spinnaker.gate.config;
+
+import com.google.common.base.Splitter;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.net.URI;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.redis.connection.jedis.JedisConnection;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.session.data.redis.config.ConfigureNotifyKeyspaceEventsAction;
+import org.springframework.session.data.redis.config.ConfigureRedisAction;
+import org.springframework.stereotype.Component;
+
+/**
+ * Runs the {@link ConfigureRedisAction} the first time a connection is retrieved.
+ *
+ * <p>This works around an issue in Spring where it tries to run the {@code ConfigureRedisAction} at
+ * startup. If Redis isn't available, the entire server will fail to start. Instead, we'll configure
+ * this setting the first time we successfully get a connection.
+ *
+ * <p>This Redis pool is used for Spring Boot's session management, not for the Rate ACL storage.
```

Comment:
What is Rate ACL?

Reply:
Sorry, most of our internal systems use the term 'rate ACL' instead of 'rate limit' (I'm not sure why), so I'm used to saying that :) Fixed.

Reply:
Cool, if you ever find out, I'd be interested to know. :)